packages/firestore/src/platform/platform.ts

```diff
@@ -60,6 +65,54 @@ export interface Platform {
   readonly base64Available: boolean;
 }
 
+/**
+ * An interface compatible with Web's ReadableStream.getReader() return type.
+ *
+ * This can be used as an abstraction to mimic `ReadableStream` where it is not
+ * available.
+ */
+export interface ByteStreamReader {
+  read(): Promise<ByteStreamReadResult>;
+  cancel(reason?: string): Promise<void>;
+}
+
+/**
+ * An interface compatible with ReadableStreamReadResult<UInt8Array>.
+ */
+export interface ByteStreamReadResult {
+  done: boolean;
+  value?: Uint8Array;
+}
+
+/**
+ * Builds a `ByteStreamReader` from a UInt8Array.
+ * @param source The data source to use.
+ * @param bytesPerRead How many bytes each `read()` from the returned reader
+ *        will read.
+ */
+export function toByteStreamReader(
+  source: Uint8Array,
+  bytesPerRead = 10240
+): ByteStreamReader {
+  let readFrom = 0;
+  return new (class implements ByteStreamReader {
```

Comment:
I assume you are using the `IIFE` pattern here to enforce the `ByteStreamReader` type?

You could just use:

const reader : ByteStreamReader = {
  ...
}
return reader;